client/lib/AwaitedEventTarget.ts

```diff
@@ -2,14 +2,14 @@ import { IJsPath } from 'awaited-dom/base/AwaitedPath';
 import IAwaitedEventTarget from '../interfaces/IAwaitedEventTarget';
 import IJsPathEventTarget from '../interfaces/IJsPathEventTarget';
 
+type IGetEventTarget = { target: Promise<IJsPathEventTarget>; jsPath?: IJsPath };
+
 export default class AwaitedEventTarget<T> implements IAwaitedEventTarget<T> {
-  constructor(
-    readonly getEventTarget: () => { target: Promise<IJsPathEventTarget>; jsPath?: IJsPath },
-  ) {}
+  constructor(readonly getEventTarget: () => Promise<IGetEventTarget> | IGetEventTarget) {}
 
   public async addEventListener<K extends keyof T>(
     eventType: K,
-    listenerFn: (this: this, event: T[K]) => any,
+    listenerFn: (...args: any[]) => any,
```

Comment:
Can this not use the same thing that IAwaitedEventTarget is using? Did you not figure this out for the typing?

Reply:
I can try to look into this if not

Reply:
I didn't figure it out. It seemed more important to move on for now.